File: java-source/src/main/java/com/template/api/TemplateApi.java
package com.template.api;

import net.corda.core.messaging.CordaRPCOps;

import javax.ws.rs.GET;
import javax.ws.rs.PUT;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.Response;
import java.util.concurrent.ExecutionException;

// This API is accessible from /api/template. The endpoint paths specified below are relative to it.
@Path("template")
public class TemplateApi {
    private final CordaRPCOps services;

    public TemplateApi(CordaRPCOps services) {
        this.services = services;
    }

    /**
     * Define GET endpoints as follows.
     */
    @GET
    @Path("templateGetEndpoint")
    @Produces(MediaType.APPLICATION_JSON)
    public void templateGetEndpoint() {}

    /**
     * Define PUT endpoints as follows.
     */
    @PUT
    @Path("templatePutEndpoint")
    public Response templatePutEndpoint(Object payload) throws InterruptedException, ExecutionException {
        return Response.accepted().build();
    }
}

File: kotlin-source/src/main/kotlin/com/template/api/TemplateApi.kt
package com.template.api

import net.corda.core.messaging.CordaRPCOps
import javax.ws.rs.GET
import javax.ws.rs.PUT
import javax.ws.rs.Path
import javax.ws.rs.Produces
import javax.ws.rs.core.MediaType
import javax.ws.rs.core.Response

// This API is accessible from /api/template. The endpoint paths specified below are relative to it.
@Path("template")
class TemplateApi(val services: CordaRPCOps) {
    /**
     * Define GET endpoints as follows.
     */
    @GET
    @Path("templateGetEndpoint")
    @Produces(MediaType.APPLICATION_JSON)
    fun templateGetEndpoint() {}

    /**
     * Define PUT endpoints as follows.
     */
    @PUT
    @Path("templatePutEndpoint")
    fun templatePutEndpoint(payload: Any): Response {
        return Response.accepted().build()
    }
}